packages/concerto-core/test/decoratormanager.js

```diff
@@ -470,4 +470,37 @@ describe('DecoratorManager', () => {
             }).should.throw(/Model violation in the "concerto.metamodel@1.0.0.Decorator" instance. Invalid enum value of "INVALID" for the field "CommandType"/);
         });
     });
-});
+
+    describe('#extractDecorators', function() {
+        it('should be able to extract decorators and vocabs from a model', async function() {
+            const testModelManager = new ModelManager({strict:true});
+            const modelText = fs.readFileSync('./test/data/decoratorcommands/test-decorated-model.cto', 'utf-8');
+            testModelManager.addCTOModel(modelText, 'test.cto');
+            let resp = DecoratorManager.extractDecorators( testModelManager);
+            let dcs = resp.decoratorCommandSet;
+            const validationModelResult = dcs.map((decoractor)=>{
+                return DecoratorManager.validate((decoractor));
+            });
+            validationModelResult.should.not.be.null;
+        });
+        it('should be able to extract decorators and vocabs from a model without namespace version', async function() {
+            const testModelManager = new ModelManager();
+            const modelTextWithoutNamespace = fs.readFileSync('./test/data/decoratorcommands/test-decorator-without-version.cto', 'utf-8');
+            testModelManager.addCTOModel(modelTextWithoutNamespace, 'test.cto');
+            const options = {
+                removeDecoratorsFromModel:true,
+                locale:'en'
+            };
+            let resp = DecoratorManager.extractDecorators( testModelManager, options);
+            let dcs = resp.decoratorCommandSet;
+            let modelManager = resp.modelManager;
+            dcs.forEach((decorator)=>{
+                modelManager=DecoratorManager.decorateModels( modelManager, decorator);
+            });
```

Comment:
Is there a way to apply a DCS atomically to a model manager?

Reply:
here, dcs is an array of decorator command set. I think we will have to iterate over each dcms and apply? Please suggest if there's an alternative.